===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ swAio_dns_lookup:n$1 == { -1 } }) free(*(swAio_dns_lookup:aio_ev)) at 5 (line 345, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/swoole-src/src/os/base.c
+++ b/table1/benchmarks/swoole-src/src/os/base.c
@@ -342,7 +342,10 @@ int swAio_dns_lookup(void *hostname, void *ip_addr, size_t size)
     aio_ev->nbytes = size;
     aio_ev->task_id = SwooleAIO.current_id++;
 
-    if (swThreadPool_dispatch(&swAioBase_thread_pool, aio_ev, sizeof(aio_ev)) < 0)
+    int tmp = swThreadPool_dispatch(&swAioBase_thread_pool, aio_ev, sizeof(aio_ev));
+    if (tmp < 0)
+      free(aio_ev);
+    if (tmp < 0)
     {
         return SW_ERR;
     }
